examples/air-purifier-app/air-purifier-common/src/air-purifier-manager.cpp

```diff
@@ -191,13 +194,75 @@ void AirPurifierManager::HandleFanControlAttributeChange(AttributeId attributeId
     }
 }
 
+void AirPurifierManager::HandleOnOff(AttributeId attributeId, uint8_t type, uint16_t size, uint8_t * value)
+{
+    if (attributeId != OnOff::Attributes::OnOff::Id)
+    {
+        return;
+    }
+    bool on = static_cast<bool>(*value);
+    uint8_t new_speed;
+    uint8_t new_percent;
+    if (on)
+    {
+        ChipLogProgress(NotSpecified, "======================== Handle ON");
+        // If either of these come back as NULL, that should mean the fan is operating in auto mode.
+        // I have no idea what that means for this case, so I'll just set them to high because this
+        // is just an example.
+        // In theory these should always be NULL together or not at all, so hopefully
+        // the checks for only percent.IsNull() or only speed.IsNull() are more theoretical
+        // than practical.
+        DataModel::Nullable<Percent> percent = GetPercentSetting();
+        DataModel::Nullable<uint8_t> speed   = GetSpeedSetting();
+        uint8_t speedMax                     = 100;
+        FanControl::Attributes::SpeedMax::Get(mEndpointId, &speedMax);
+        if (percent.IsNull() && speed.IsNull())
+        {
+            // Operating in auto mode, set to 100
+            new_percent = 100;
+            new_speed   = speedMax;
+        }
+        else if (percent.IsNull())
+        {
+            // This should never happen, but nonetheless, let's check and warn.
+            ChipLogError(NotSpecified,
+                         "AirPurifierManager::HandleOnOff: PercentSetting is null when SpeedSetting is not. Setting PercentCurrent "
+                         "from Speed");
+            new_speed   = speed.Value();
+            new_percent = static_cast<uint8_t>(new_speed * 100u / speedMax);
```

Comment:
If speedMax is 0 by mistake, you have a divide by zero. Consider setting speedMax at line 217 to `0` and basically handling nothing of speeds at all if speedmax is zero?

Reply:
Yup - good catch. Added a check up top. Thanks.